port/win/port_win.cc

```diff
@@ -142,7 +142,7 @@ DIR* opendir(const char* name) {
     return nullptr;
   }
 
-  strncpy_s(dir->entry_.d_name, dir->data_.name, strlen(dir->data_.name));
+  strcpy_s(dir->entry_.d_name, sizeof(dir->entry_.d_name), dir->data_.name);
```

Comment:
I see, but not all users have the latest and greatest mingw, and it makes sense to support at least one version back. Moreover, using `strncpy` with `strlen` is funny.